ipa-core/src/protocol/ipa_prf/prf_sharding/mod.rs

```diff
@@ -505,13 +505,14 @@ where
             * multiplications_per_record::<BK, TV, TS>(attribution_window_seconds));
 
     // Tricky hacks to work around the limitations of our current infrastructure
-    let num_outputs = input_rows.len() - histogram[0];
     let mut dzkp_validator = sh_ctx.clone().dzkp_validator(
         MaliciousProtocolSteps {
             protocol: &Step::Attribute,
             validate: &Step::AttributeValidate,
         },
-        chunk_size,
+        // The size of a single batch should not exceed the active work limit,
+        // otherwise it will stall
+        std::cmp::min(sh_ctx.active_work().get(), chunk_size),
```

Comment:
@andyleiserson does this change make sense to you? I had to change it because otherwise I was getting stalls even on very small inputs, when `active_work` is larger than `chunk_size`.

Reply:
Is this expected to interact with #1307? I would expect active work (even at 32k) to be quite a bit less than the target proof size. I believe there are only ~50 multiplications per record in attribution? Which would mean a chunk size of 1M for a proof size of 50M.

Reply:
Sort of. If the chunk size is larger than active work in this case(I think it may only apply to `validated_seq_join` though), then we are going to stall, because we won't schedule enough futures to fill up the batch and `validate_record` will prevent `seq_join` from scheduling more futures.

so it is crucial that `validated_seq_join` populates the whole batch and validates it inside one active work window

Reply:
Oof yeah. I feel like we talked about this in July and then I clearly forgot about it.

I just looked at the original malicious share conversion code (before all the `validate_record` stuff) and it has two seq_joins, one before the reveal, and one after the reveal.